lib/cloudant.js

```diff
@@ -524,21 +531,52 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
           model, where, data, options);
   var self = this;
   var mo = self.selectModel(model);
+  var idName = self.idName(model);
+  // if idName exists, change the idName to be the default `_id` key
+  if (idName && data[idName]) {
+    data._id = data[idName];
+    delete data[idName];
+  }
+  var res = {};
+  var docList = [];
+
   self.all(model, {where: where}, {raw: true}, function(err, docs) {
     if (err) return cb(err, docs);
     if (docs.length === 0) return cb(null, {count: 0});
 
     data = getPlainJSONData.call(self, model, data);
+
+    /**
+     * for each of the docs from the given filter, only update the instance where the revs match
+     * behavior of bulk update in cloudant:
```

Comment:
First things first, `data` passed to the `update` function is an **object** not an **array of object**.

Being said that, the code I have right now deletes the `_id` and `_rev` out of each of the docs we get back **iff** `doc._id === data._id || doc._rev === data._rev` and then merges the data with the doc. 

Upon that condition, it also adds to the an array of docLists that needs to be updated and on the final callback it calls `mo.db.bulk({docs: docList}, function(err, result) {...});`

On the callback, we check for the result and if there are any elements in the result array that has `error`, we throw the error. Else we just return the count of the number of result updated.